docs/workshops/advanced/node-workflow.md

```diff
@@ -9,54 +9,54 @@ begin
+To see the actual step code, go to `workflows/node/node_create.py`. Covering the implementation steps of these details is generally out of scope for this workshop, they simply exist to give us a framework to build off of. The important concept to understand is that we are going to fill out the currently empty `set_node_to_active` step to actually perform the functions we need done in netbox. Most of the heavy lifting with netbox is performed in the netbox single dispatch service we've included with the workshop code at `services/netbox.py` and `products/services/netbox/netbox.py`. To keep things simple, we won't cover _how_ this single dispatch service works, but rather, just use it. Generally, to use this service, you pass the subscription instance that you want to update in netbox to the `build_payload()` function and netbox will be updated for you, as shown in the last step of the node create workflow:
+
+```python
+@step("Update Node in Netbox")
+def update_node_in_netbox(subscription: NodeProvisioning) -> State:
+    """Updates a node in Netbox"""
+    netbox_payload = build_payload(subscription.node, subscription)
+    return {"netbox_payload": netbox_payload.dict(), "netbox_updated": netbox.update(netbox_payload)}
+
+```
 
 ### Why Update Node Status Externally?
 
-The goal of the orchestrator is generally not to store data in the orchestrator's DB, but rather, to store pointers to data in external systems and keep track of those data. With the way that `create_node` is currently implemented, we have a bug where it is possible to enroll the same node multiple times, due to how we are filtering our list of nodes. To see this bug, take a look at the `get_nodes_list()` function in `create_node.py`, copied here for your convenience:
+The goal of the orchestrator is generally not to store data in the orchestrator's DB, but rather, to store pointers to data in external systems and keep track of those data. With the way that `create_node` is currently implemented, we have a bug where it is possible to enroll the same node multiple times, due to how we are filtering our list of nodes. To see this bug, take a look at the first few lines of the `initial_input_form_generator` function in `create_node.py`, copied here for your convenience:
 
 ```python
-def get_nodes_list() -> List[Any]:
-    """
-    Connects to netbox and returns a list of netbox device objects.
-    """
-    logger.debug("Connecting to Netbox to get list of available nodes")
-    node_list = list(netbox.dcim.devices.filter(status="planned"))
-    logger.debug("Found nodes in Netbox", amount=len(node_list))
-    return node_list
+def initial_input_form_generator(product_name: str) -> FormGenerator:
+    """Generates the Node Form to display to the user."""
+    logger.debug("Generating initial input form")
+    nodes = netbox.get_devices(status="planned")
```

Comment:
It may be clearer to rename `nodes` to `devices` to emphasise `netbox.get_devices()` is returning a list of Netbox `Devices` objects and not a list of node subscriptions.

Reply:
Made the changes myself :-D